gitpod-io/gitpod
# [ws-manager] Add client-go prometheus metrics

The new metrics are defined in [controller-runtime](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/metrics/client_go_adapter.go)

**How to test:**
- ensure the werft build is green
- ensure the in-workspace build works
- check new ws-manager metrics are available running ` kubectl port-forward pod/ws-manager-.... 9500`
- open browser port `9500` path `/metrics`

The output should contain metrics like:
```
rest_client_request_latency_seconds_bucket{url="https://10.63.240.1:443/api/v1/namespaces/%7Bnamespace%7D/services?labelSelector=%7Bvalue%7D",verb="GET",le="0.256"} 25
rest_client_request_latency_seconds_bucket{url="https://10.63.240.1:443/api/v1/namespaces/%7Bnamespace%7D/services?labelSelector=%7Bvalue%7D",verb="GET",le="0.512"} 25
rest_client_request_latency_seconds_bucket{url="https://10.63.240.1:443/api/v1/namespaces/%7Bnamespace%7D/services?labelSelector=%7Bvalue%7D",verb="GET",le="+Inf"} 25
rest_client_request_latency_seconds_sum{url="https://10.63.240.1:443/api/v1/namespaces/%7Bnamespace%7D/services?labelSelector=%7Bvalue%7D",verb="GET"} 0.119781686
rest_client_request_latency_seconds_count{url="https://10.63.240.1:443/api/v1/namespaces/%7Bnamespace%7D/services?labelSelector=%7Bvalue%7D",verb="GET"} 25
# HELP rest_client_requests_total Number of HTTP requests, partitioned by status code, method, and host.
# TYPE rest_client_requests_total counter
rest_client_requests_total{code="200",host="10.63.240.1:443",method="DELETE"} 79
rest_client_requests_total{code="200",host="10.63.240.1:443",method="GET"} 2047
rest_client_requests_total{code="200",host="10.63.240.1:443",method="PUT"} 318
rest_client_requests_total{code="201",host="10.63.240.1:443",method="POST"} 80
rest_client_requests_total{code="404",host="10.63.240.1:443",method="DELETE"} 80
rest_client_requests_total{code="404",host="10.63.240.1:443",method="GET"} 1326
```

After #3208 the metrics will include additional reflectors information

xref: #3208
